include/experimental/__p0009_bits/fwd.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+namespace MDSPAN_IMPL_STANDARD_NAMESPACE {
+namespace detail {
+
+// type used for rank-based tag dispatch
+//
+// this is used to enable alternatives to constexpr if when building for C++14
+//
+template <std::size_t N>
+struct with_rank {};
```

Comment:
Would you consider making this an alias of `std::integral_constant`, so as to avoid more instantiations of types?